### DISCUSSION TITLE
always update changelog.md

### DISCUSSION THREAD

**Discussion by huczas** [semantic-release/semantic-release#3552] [Configuration] [answered]
I can’t figure out how to ensure that the changelog.md file is always updated even if change is not bumping my version.
For example, I have a branch called develop, which is my default branch.
If I merge a feature branch with a commit message starting with “feat: some new change,” it should immediately appear in my changelog file under the “Features” section and it is also bumping version with minor as expected, also if I add more commits all is ok.

However, if I merge a branch with only one  commit message like “ci: changed pipeline,” this change is ignored, and changelog.md is not updated. How to force update changelog.md to always update?

below is my config:

```json
{
  "branches": ["develop"],
  "preset": "conventionalcommits",
  "tagFormat": "${version}",
  "plugins": [
    [
      "@semantic-release/commit-analyzer",
      {
        "preset": "conventionalcommits",
        "releaseRules": [
          { "type": "breaking", "release": "major" },
          { "type": "major", "release": "major" },
          { "type": "feature", "release": "minor" },
          { "type": "feat", "release": "minor" },
          { "type": "fix", "release": "patch" },
          { "type": "bugfix", "release": "patch" },
          { "type": "hotfix", "release": "patch" },
          { "type": "patch", "release": "patch" }
        ]
      }
    ],
    [
      "@semantic-release/release-notes-generator",
      {
        "preset": "conventionalcommits",
        "presetConfig": {
          "types": [
            { "type": "milestone", "section": "✨ Milestone", "hidden": false },
            { "type": "breaking", "section": "🧨 Breaking changes!", "hidden": false },
            { "type": "feat", "section": "🚀 Features", "hidden": false },
            { "type": "feature", "section": "🚀 Features", "hidden": false },
            { "type": "improv", "section": "💎 Improvements", "hidden": false },
            { "type": "bugfix", "section": "🛠 Fixes", "hidden": false },
            { "type": "fix", "section": "🛠 Fixes", "hidden": false },
            { "type": "perf", "section": "⏩ Performance", "hidden": false },
            { "type": "refactor", "section": "✂️ Refactor", "hidden": false },
            { "type": "revert", "section": "👀 Reverts", "hidden": false },
            { "type": "style", "section": "💈 Style", "hidden": false },
            { "type": "test", "section": "🧪 Tests", "hidden": false },
            { "type": "ci", "section": "👾 CI/CD", "hidden": false },
            { "type": "infr", "section": "🔩 Infrastructure", "hidden": false },
            { "type": "docs", "section": "📔 Docs", "hidden": false },
            { "type": "chore", "section": "📦 Other", "hidden": false }
          ]
        },
        "writerOpts": {
          "groupBy": "type"
        },
        "parserOpts": {
          "noteKeywords": ["BREAKING CHANGE", "BREAKING CHANGES", "BREAKING"]
        },
        "linkCompare": true,
        "linkReferences": true
      }
    ],
    [
      "@semantic-release/changelog",
      {
        "changelogFile": "docs/CHANGELOG.md"
      }
    ],
    [
      "@semantic-release/git",
      {
        "message": "Changelog updated: ${nextRelease.version} [skip ci]",
        "assets": ["docs/CHANGELOG.md"]
      }
    ],
    [
      "semantic-release-ado",
      {
        "varName": "SemVer",
        "setOnlyOnRelease": true,
        "isOutput": true
      }
    ]
  ]
}
```

**Comment by travi** [comment, marked as answer]
keep in mind that semantic-release is not a generic workflow tool. it is named semantic-release because it is specifically for enabling an easier approach to releasing semantically. the semantic meaning of the version bump for a release is meant to communicate the impact of the changes contained in that release to your consumers. 

> if I merge a branch with only one commit message like “ci: changed pipeline,” this change is ignored

this commit is ignored because semantically, it has no impact for the consumers of your library. there is no reason to release a new version for a change that does not impact your consumers. it is possible to override this behavior and trigger a release for a "ci" commit type by adding it under `releaseRules` for the commit-analyzer plugin, but we dont recommend that for the reasons above. also, the changelog plugin only updates the changelog file when a release happens 

in addition, we [recommend against using the changelog plugin](https://semantic-release.gitbook.io/semantic-release/support/faq#should-release-notes-be-committed-to-a-changelog.md-in-my-repository-during-a-release) at all, if you can avoid it. if you are using github releases, or similar for ado (i'm less familiar with what is available there), you can [significantly reduce the complexity of your release](https://semantic-release.gitbook.io/semantic-release/support/faq#making-commits-during-the-release-process-adds-significant-complexity) by avoiding the commit needed to update that file.

**Comment by huczas** [comment]
after some tests i see that my ci commits with changes are added in next version bump that is ok fore me. Thanks for clarifying.

**Reply by muratkarakas** [reply]
can you please share your setup and process/command. I have similar issue, can't managed to add docs commits to changelog.md file. It only adds feat and fix commits.  Here is my config

`
module.exports = {
  branches: ["main", "release/v*"],

  "dryRun": false,
  plugins: [
    ["@semantic-release/commit-analyzer", {
      "releaseRules": [
        { "type": "feat", "release": "minor" },
        { "type": "fix", "release": "patch" },
        { "type": "revert", "release": "patch" },
      ]
    }],
    ["@semantic-release/changelog", {
      "changelogFile": "CHANGELOG.md"
    }],
    ['@semantic-release/release-notes-generator', {
      presetConfig: {
        types: [
          { type: 'feat', section: 'Features' },
          { type: 'fix', section: 'Bug Fixes' },
          { type: 'perf', section: 'Performance Improvements' },
          { type: 'revert', section: 'Reverts' },
          { type: 'docs', section: 'Documentation',hidden: false}, // Docs commit'leri için ayrı bir bölüm
          { type: 'test', section: 'Tests' }, // Test commit'leri için ayrı bir bölüm
          { type: 'build', section: 'Build System', hidden: false },
          { type: 'ci', section: 'Continuous Integration', hidden: false },
          { type: 'chore', section: 'Miscellaneous', hidden: false },
        ]
      }
    }],

    ["@semantic-release/git", { // Güncellenmiş dosyaları commit'ler ve tag'leri push eder
      "assets": ["CHANGELOG.md", "package.json", "package-lock.json"],
      "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}",
      "repositoryUrl": `https://gitlab-ci-token:${process.env.CI_JOB_TOKEN}@${process.env.CI_SERVER_HOST}/${process.env.CI_PROJECT_PATH}.git`
    }],
  ]
};



`

**Reply by huczas** [reply]
build-up this part with relevant for you types:

["@semantic-release/commit-analyzer", {
"releaseRules": [
{ "type": "feat", "release": "minor" },
{ "type": "fix", "release": "patch" },
{ "type": "revert", "release": "patch" },
]
}],

**Reply by muratkarakas** [reply]
Thank you
